### DISCUSSION TITLE
Updating to v14+

### DISCUSSION THREAD

**Discussion by jurijsk** [nlp-compromise/nlp_compromise#948] [Q&A]
I've update the version of compromise to v14 something then checked the change log which says:
`
[breaking] - remove .nouns().toPossessive() & .nouns().hasPlural()
`
with out offering any replacement for the methods.

I wonder if there is a replacement for whose methods via plugins or other means?

**Comment by spencermountain** [comment]
hey Jurijs, it's great that you found a use for those methods - i didn't know anyone was using them.
Yeah, i'm happy to bring them back.

you can replicate `.hasPlural()` by checking for the `#Uncountable` tag:
```js
nlp.extend({
  api: (View) => {
    View.prototype.hasPlural = function () {
      return this.nouns().filter(m => m.has('#Plural') || !m.has('#Uncountable'))
    }
  }
})

doc = nlp('peace, a flower, and doves').debug()
doc.nouns().hasPlural().debug()
// a flower, doves
```

out of curiosity, what is your use-case for these methods? 
maybe they could be done in a more-helpful way. cheers
